src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
```

Comment:
Consider validating the `data` parameter to ensure it's not null or empty before proceeding with the conversion.